goal_src/levels/sunken/puffer.gc

```diff
@@ -5,3 +5,1688 @@
+    )
+   (else
+    (let ((v1-21 (-> obj draw shadow-ctrl)))
+     (logior! (-> v1-21 settings flags) 32)
+     )
+    0
+    )
+   )
+  (none)
+  )
+
+(defmethod dummy-24 puffer ((obj puffer) (arg0 vector))
+  (and
+   (dummy-25 (-> obj nav) arg0 11468.8)
+   (<
+    (-> arg0 y)
+    (+ (-> obj root-override trans y) (-> obj fact-info-override notice-top))
+    )
+   )
+  )
+
+(defmethod dummy-22 puffer ((obj puffer))
+  (let* ((a1-0 (-> obj buddy))
+         (v1-0 (if a1-0
+                (-> a1-0 ppointer 3)
+                )
+          )
+         )
+   (if
+    (and
+     v1-0
+     (>=
+      25395.2
+      (vector-vector-xz-distance
+       (-> obj root-override trans)
+       (-> (the-as process-drawable v1-0) root trans)
+       )
+      )
+     )
+    (return #t)
+    )
+   )
+  #f
+  )
+
+(defmethod dummy-25 puffer ((obj puffer) (arg0 float))
+  (when *target*
+   (let ((gp-0 (target-pos 0)))
+    (when
+     (and
+      (zero? (logand (-> *target* state-flags) #x80f8))
+      (dummy-24 obj gp-0)
+      (>= (-> gp-0 y) (+ -14336.0 (-> obj attack-bottom-y)))
+      (>= (+ 2048.0 (-> obj top-y)) (-> gp-0 y))
+      )
+     (let
+      ((f30-0 (vector-vector-xz-distance gp-0 (-> obj root-override trans))))
+      (when (>= arg0 f30-0)
+       (let* ((a0-4 (-> obj buddy))
+              (v1-12 (if a0-4
+                      (-> a0-4 ppointer 3)
+                      )
+               )
+              )
+        (cond
+         (v1-12
+          (if (not (-> (the-as puffer v1-12) attacking?))
+           (return #t)
+           )
+          (if
+           (<
+            f30-0
+            (vector-vector-xz-distance
+             gp-0
+             (-> (the-as puffer v1-12) root-override trans)
+             )
+            )
+           (return #t)
+           )
+          )
+         (else
+          (return #t)
+          )
+         )
+        )
+       )
+      )
+     )
+    )
+   )
+  #f
+  )
+
+(deftype pick-patrol-point-away-from-buddy-work (structure)
+  ((best-path-index int32          :offset-assert   0)
+   (best-rating     float          :offset-assert   4)
+   (best-dest       vector :inline :offset-assert  16)
+   (pt-dir          vector :inline :offset-assert  32)
+   (buddy-dir       vector :inline :offset-assert  48)
+   (dest            vector :inline :offset-assert  64)
+   )
+  :method-count-assert 9
+  :size-assert         #x50
+  :flag-assert         #x900000050
+  )
+
+
+(defmethod dummy-23 puffer ((obj puffer) (arg0 symbol))
+  (local-vars (v1-0 process))
+  (set! v1-0 (when arg0
+              (let ((a0-1 (-> obj buddy)))
+               (set! v1-0 (if a0-1
+                           (-> a0-1 ppointer 3)
+                           )
+                )
+               )
+              (if (not v1-0)
+               (set! arg0 #f)
+               )
+              v1-0
+              )
+   )
+  (cond
+   (arg0
+    (let ((s4-0 (-> obj path curve num-cverts))
+          (s5-0 (new 'stack-no-clear 'matrix))
```

Comment:
I'm not sure what the right type is here, but I don't think it's `matrix`

Reply:
i just made an `inline-array` of 5 `vector`s and it seems to have worked out.